src/driver/xdma_sgdma.c

```diff
@@ -33,6 +33,20 @@ unsigned int sgdma_timeout = 10;
 module_param(sgdma_timeout, uint, 0644);
 MODULE_PARM_DESC(sgdma_timeout, "timeout in seconds for sgdma, default is 10 sec.");
 
+struct xdma_channel *xdma_get_c2h(struct hermes_pci_dev *hpdev)
+{
+	if (hpdev->c2h_channel_max)
+		return &hpdev->xdma_c2h_chnl[0];
```

Comment:
What prevents two users from taking the same channel?

Why would we support c2h_channel_max == 0.... It should really fail to initialize if that's the case, not fail when someone tries to use it.

Reply:
Hmm yes... The xdma driver already fails to initialize when c2h_channel_max == 0, so we don't need this check. Also, the XDMA driver already blocks when multiple request are sent to the same channel, so I'm going to always return channel 0 for now